docs/advanced-topics/negative-mining.md

```diff
@@ -0,0 +1,86 @@
+(negative-mining)=
+# {octicon}`telescope` Negative Mining
+
+## Context: Deep Metric Learning
+
+Metric Learning algorithms attempt to teach the Deep Neural Network (DNN) to tell
+which objects are semantically/visually similar and which ones are not.
+
+For uni-modal fine-tuning tasks such as text-to-text, image-to-image, or mesh-to-mesh,
+Finetuner is dependent on algorithms & tricks developed for metric learning.
+
+![batch-sample](../imgs/batch-sampling.png)
+
+Assume we have a list of Documents belonging to four classes: `1`, `2`, `3`, and `4`,
+Finetuner will evenly sample *X* items per class to fulfill a batch *B*.
+
+In the next step,
+Finetuner creates all possible Triplets *(anchor, pos, neg)* from a batch.
+These Triplets become the training data.
+The objective is to pull Documents that belong to the same class together,
+while pushing the Documents which belong to a different class away from each other.
+
+![training](../imgs/metric-train.png)
+
+
+## Why mining?
+
+After Triplets construction,
+all possible *(anchor, pos, neg)* Triplets becomes the training data.
+However,
+what if the pre-trained model already performs well on some of the Triplets?
+
+To put it in another way: *what if the distance between `anchor` and `pos` is smaller than
+the distance between `anchor` and `neg`?*
+
+These Triplets,
+does not contribute to improving the model.
+A more effective way is to use hard/semi-hard negative samples for model training.
```

Comment:
These three paras are not well constructed. I need to think about how to say it.

Reply:
TODO: to short to understand it

Reply:
maybe move this mining part to advanced configuration